### PR TITLE
Add TheHive connector for cases

### PR DESCRIPTION
**Pull request by lcawl**
Relates to https://github.com/elastic/kibana/pull/180931

This PR updates https://www.elastic.co/guide/en/security/current/cases-manage-settings.html#cases-ui-integrations and https://www.elastic.co/docs/current/serverless/security/cases-settings#external-incident-management-systems with support for TheHive connector in cases.

It also removes the table from the "Mapped case fields" section of those pages, since that reference information was outdated (e.g. missing the tags fields) and does not exist in the other solution-specific copies of this page.  If that information is necessary, the most accurate source of truth seems to be the `mapping.ts` file in each connector folder under `x-pack/plugins/cases/server/connectors`.

### Preview

- https://security-docs_bk_5859.docs-preview.app.elstc.co/guide/en/security/master/cases-manage-settings.html#cases-ui-integrations
- https://elastic-dot-co-docs-production-fz7jknxut-elastic-dev.vercel.app/current/serverless/security/cases-settings#external-incident-management-systems

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-bpeh9qm3i-elastic-dev.vercel.app

* Built with commit 93ba2ee162fb279389020b6c4d718b49e3e8bb46

> Issues? Visit #next-docs in Slack

File: docs/cases/cases-manage-settings.asciidoc [hunk 1]
```diff
@@ -42,13 +43,14 @@ https://www.elastic.co/subscriptions[appropriate license], and your role needs *
 To create a new connector:
 
 . From the *Incident management system* list, select *Add new connector*.
-. Select the system to send cases to: *{sn}*, *{jira}*, *{ibm-r}*, *{swimlane}*, or *{webhook-cm}*.
+. Select the system to send cases to: *{sn}*, *{jira}*, *{ibm-r}*, *{swimlane}*, *TheHive*, or *{webhook-cm}*.
```
Comment (nastasha-solomon) [inline docs/cases/cases-manage-settings.asciidoc:46]: Should we make a shared attribute for this, or is it unnecessary in this case?

**Review by nastasha-solomon — APPROVED** [review]
Left one question, but otherwise everything LGTM! Thanks again for updating the Security docs!

File: docs/cases/cases-manage-settings.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/cases/cases-manage-settings.asciidoc:46]: It doesn't hurt!  I created https://github.com/elastic/docs/pull/3076 and https://github.com/elastic/docs-site/pull/45

File: docs/serverless/investigate/cases-settings.mdx [hunk 2]
```diff
@@ -40,7 +41,7 @@ To create a new connector
 
 1. From the **Incident management system** list, select **Add new connector**.
 
-1. Select the system to send cases to: **((sn))**, **((jira))**, **((ibm-r))**, **((swimlane))**, or **((webhook-cm))**.
+1. Select the system to send cases to: **((sn))**, **((jira))**, **((ibm-r))**, **((swimlane))**, **TheHive**, or **((webhook-cm))**.
    ![Shows the page for creating connectors](../images/cases-settings/security-cases-connectors.png)
```
Comment (joepeeples) [inline docs/serverless/investigate/cases-settings.mdx:45]: Could we omit this screenshot? It's not included in the ESS/AsciiDoc docs, and doesn't seem like essential information for the user.

Comment (lcawl) [reply docs/serverless/investigate/cases-settings.mdx:45]: One less screenshot to maintain works for me!

File: docs/cases/cases-manage-settings.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/cases/cases-manage-settings.asciidoc]: Fixed in fbcf2423c161147e0e3fb3912ad26a59e18805ab

File: docs/serverless/investigate/cases-settings.mdx (hunk 2 above)
Comment (lcawl) [reply docs/serverless/investigate/cases-settings.mdx]: Fixed in 98fcda77c6ef4a71c58a90fb75b7eccfb300a6e1

**Review by joepeeples — APPROVED** [review]

**Pull request merged**